### DISCUSSION TITLE
MQTT connection issue

### DISCUSSION THREAD

**Discussion by samalitu** [collin80/SavvyCAN#364] [Q&A] [answered]
Hi.

I've been trying to setup an mqtt connection, but even though i've specified the broker information in the settings, and the topic in the connection initialization, it always just says "not connected". I tried writing the topic in different forms, as i thought that might've been the issue but with no luck. I tried this on both windows 10 and linux mint, with the same results. 

Has anyone else encountered this? and if so, has a solution to it been found yet? 

![settings](https://user-images.githubusercontent.com/54707941/122016992-dc19e980-cdc1-11eb-8213-5714d27c3b70.png)
![connection settings](https://user-images.githubusercontent.com/54707941/122017000-dde3ad00-cdc1-11eb-9072-3cc045ff7c60.png)

**Comment by collin80** [comment]
In all the testing I did I was using SSL and a user name and password. So, it is entirely possible that it currently requires that specific set up. Further testing will have to be done with your configuration - no user name or password required, no SSL.

**Comment by collin80** [comment, marked as answer]
I just pushed a commit that might fix the problem for you. If it doesn't then it still should allow you to debug what is happening better. Enable the console (check box) before you do the MQTT connection and it should output text into the console area that helps to figure out why it isn't working. You'll have to pull the code and compile it yourself or wait for CI/CD to finish baking and grab the new binary.

**Reply by carlsa** [reply]
Thank you!

I have tested the c491be7 CI build on Windows with these results:

- Unencrypted without authentication - ✅ **Works**
- Unencrypted with authentication - ✅ **Works**
- Encrypted without authentication - ❌ **Fails with message** `MQTT SSL Internal Error`
- Encrypted with authentication - ❌ **Fails with message** `MQTT Remote Host Closed The Connection`

I'm using the [test servers from the Mosquitto project](https://test.mosquitto.org/).

I think the _encrypted with authentication_ case fails since the test server isn't on port 8883.

I couldn't get the CI build to run on Linux, there it fails with the message below. The release builds work fine however.
```
SavvyCAN-x86_64.AppImage: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by SavvyCAN-x86_64.AppImage)
```

**Reply by collin80** [reply]
You're right, currently there is no checkbox or anything to select for SSL so it only does SSL if the port matches 8883. Instead there should be a checkbox to select SSL and then it should allow connecting on any port.

What version of LINUX are you using? I like to upgrade as soon as possible so I'm running Ubuntu 21.04 and it has GLIBC 2.33 which seems to be acceptable to the AppImage. I just tested it to work. But, if you are running something that still uses GLIBC 2.28 or lower then I can see this would be a problem. For instance, by default Ubuntu 18.04 does not have a recent enough GLIBC to work. It might be possible for me to select an older linux version for the CI builder.
